app/src/main/java/com/rkm/tasky/sync/worker/AgendaSyncWorker.kt

```diff
@@ -0,0 +1,113 @@
+package com.rkm.tasky.sync.worker
+
+import android.content.Context
+import androidx.hilt.work.HiltWorker
+import androidx.work.CoroutineWorker
+import androidx.work.WorkerParameters
+import com.rkm.tasky.database.TaskyDatabase
+import com.rkm.tasky.database.dao.AgendaDao
+import com.rkm.tasky.database.model.AgendaDetails
+import com.rkm.tasky.database.model.AttendeeEntity
+import com.rkm.tasky.database.model.EventAttendeesAndPhotosToDelete
+import com.rkm.tasky.database.model.PhotoEntity
+import com.rkm.tasky.di.ApplicationCoroutineScope
+import com.rkm.tasky.di.IoDispatcher
+import com.rkm.tasky.network.datasource.TaskyAgendaRemoteDataSource
+import com.rkm.tasky.network.util.NetworkError
+import com.rkm.tasky.network.util.safeCall
+import com.rkm.tasky.repository.mapper.asEventWithDetails
+import com.rkm.tasky.repository.mapper.asReminderEntity
+import com.rkm.tasky.repository.mapper.asTaskEntity
+import com.rkm.tasky.util.result.Result
+import com.rkm.tasky.util.result.onFailure
+import dagger.assisted.Assisted
+import dagger.assisted.AssistedInject
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.withContext
+
+@HiltWorker
+class AgendaSyncWorker @AssistedInject constructor(
+    @Assisted context: Context,
+    @Assisted params: WorkerParameters,
+    private val agendaRemoteDataSource: TaskyAgendaRemoteDataSource,
+    private val agendaLocalDataSource: AgendaDao,
+    private val db: TaskyDatabase,
+    @ApplicationCoroutineScope private val scope: CoroutineScope,
+    @IoDispatcher private val dispatcher: CoroutineDispatcher,
+): CoroutineWorker(context, params) {
+    override suspend fun doWork(): Result  = withContext(scope.coroutineContext + dispatcher){
+        val agendaResults = safeCall { agendaRemoteDataSource.getFullAgenda() }
+
```

Comment:
Your code here is still cancellable if you use `withContext(appScope.coroutineContext)`. Although withContext executes the code in the appScope’s context, it is not an independent coroutine like when `appScope.launch{}` is called. 

But here, it is not as crucial since this code runs in the background (compared to the viewModelScope, which is cancelled when the VM is cleared).